modules/core/shared/src/main/scala/lucuma/core/model/Target.scala

```diff
@@ -35,12 +36,13 @@ object Target extends WithId('t') with TargetOptics {
     name:          NonEmptyString,
     tracking:      SiderealTracking,
     sourceProfile: SourceProfile,
+    catalogInfo:   Option[CatalogInfo],
```

Comment:
I suppose there is no standard for the values we might see in order to enumerate them.  I wonder if we could parse these values when possible into something we could use.  In the mockups we show, for example, galaxies with a unique icon in target lists.

Reply:
The object types of sidereal objects would not overlap with those of nonsidereal objects, but if it is just a `String` that we display then it seems fine to put this on `Target`.  If we could dig out what the types actually are though, then it might make more sense to separate the sidereal types from the nonsidereal ones and have `SiderealObjectType` and `NonSiderealObjectType`.

Reply:
In the case of Simbad, the requirement here is to present to the user the object type, morphological type and spectral type, so that they can use this to fill out the SED fields. Something along these lines in the mockup:

<img width="400px" src="https://user-images.githubusercontent.com/1895643/146978245-a27ab67d-8254-452e-bf78-9a2f3f2e3c73.png"/>

So for Simbad, we would concatenate those 3 fields into a single string and put it as `objectType`, just to be able to present it to the user. For the moment, we are not extracting semantic information out of this. The UI could actually fetch this info from Simbad when needed, since we are storing the catalog id. So, this actually acts as a cache to avoid that extra query to Simbad.

I moved `CatalogInfo` to `Target` since I'm expecting some kind of object type string to come from Horizons too, but this seems to add redundance with the ephemeris key. We should probably move `CatalogInfo` into `Target.Sidereal` then. I'll make that change. I'll also change the `objectType` to an `Option[NonEmptyString]`. 

I agree it would be nice to have a nicely modeled object type on the target and show this as an icon, but that's not what this PR attempts to address. When we do have such field, the parser can make a best-effort attempt to infer it based on the catalog's object type info. Also, the plan is for the parser to eventually attempt to infer the SED from the catalog object type.